drizzlepac/alignimages.py

```diff
@@ -82,45 +82,72 @@ def check_and_get_data(input_list,**pars):
 
+    regtest: Boolean
+        Flag to signal the use of PyTest to run regresssion tests.  When using PyTest with individual images
+        which are part of an association (e.g., *_flt.fits), the files cannot be obtained from MAST as 
+        singletons. They are stored in Artifactory or local disk as part of the regression test suite.  
+
+    test_data_path : List of strings  
+        List of comma separated strings defining the path on Artifactory or local disk where the *_fl[t|c].fits 
+        files are stored.  The root of the path is defined by the environment variables TEST_BIGDATA.
+
+
     Returns
     =======
     input_file_list : list
         list of full filenames
 
     """
     totalInputList = []
-    retrieve_list = [] # list of already retrieved ASNs, only request each once
-    for input_item in input_list:
-        try:
-            filelist = aqutils.retrieve_observation(input_item,**pars)
-        except Exception:
-            filelist = []
-        if len(filelist) == 0:
-            # look for local copy of the file
-            input_rootname = input_item[:input_item.find('_')]
-            fitsfilenames = sorted(glob.glob("{}_fl?.fits".format(input_rootname)))
-            if len(fitsfilenames) > 0:
-                imghdu = fits.open(fitsfilenames[0])
-                imgprimaryheader = imghdu[0].header
-                try:
-                    asnid = imgprimaryheader['ASN_ID'].strip().lower()
-                    if asnid == 'NONE':
-                        asnid = None
-                except KeyError:
-                    asnid = None
-                if asnid and asnid not in retrieve_list:
+
+    # If the regression test flag is True, a regression test has been invoked where the input_list 
+    # actually contains the full filename(s) (e.g., *_flt.fits).  Note that not all regression tests use 
+    # this functionality as Artifactory or a local disk is accessed for the data source with the use of 
+    # this flag.  Some regression tests use astroquery to connect to MAST. This regression testing code 
+    # is not a scalable solution, but it is necessary for the alignment portion of the drizzle software.
+    if pars['regtest'] == True:
+        # Test folder containing the data - the root path is defined by the environment variable TEST_BIGDATA
+        data_path = pars['test_data_path']
+        from ci_watson.artifactory_helpers import get_bigdata
+        for infile in input_list:
+            full_file_path = get_bigdata(data_path[0],data_path[1],data_path[2],infile)
+            totalInputList.append(infile)
+    else:
+        archive = pars['archive']
+        clobber = pars['clobber']
+        retrieve_list = [] # list of already retrieved ASNs, only request each once
+        for input_item in input_list:
+            try:
+                filelist = aqutils.retrieve_observation(input_item,archive=archive,clobber=clobber)
+            except Exception:
```

Comment:
`except Exception:` is the same as writing `except:`. If this `try/except` block is required due to how `aqutils.retrieve_observation` works, then the exception raised should be trapped explicitly (i.e. `except aqutils.OhNoes:`).

Reply:
In reading the Pythonic way to handle the exception clause, I see what you are suggesting.  In our case I really do not know all the possible exceptions which can be raised.  It seems in these cases it is acceptable to modify the except block to log the full stack trace.  I have done this in the PyTest routines, but there may be conflicts with the functionality of PyTest so I am familiar with all the goodness I get.  In this case, I would change the exception clause to be 

    except Exception:
        exc_type, exc_value, exc_tb = sys.exc_info()
        traceback.print_exception(exc_type, exc_value, exc_tb, file=sys.stdout)
        filelist = []